Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadInstanceUnpacker.cs

```diff
@@ -0,0 +1,162 @@
+      // using the normal def reference.
+      ObjectId definitionId = !instance.AnonymousBlockTableRecord.IsNull
+        ? instance.AnonymousBlockTableRecord
+        : instance.BlockTableRecord;
+
+      InstanceProxy instanceProxy =
+        new()
+        {
+          applicationId = instanceId,
+          definitionId = definitionId.ToString(),
+          maxDepth = depth,
+          transform = GetMatrix(instance.BlockTransform.ToArray()),
+          units = _unitsConverter.ConvertOrThrow(Application.DocumentManager.CurrentDocument.Database.Insunits)
+        };
+      _instanceObjectsManager.AddInstanceProxy(instanceId, instanceProxy);
+
+      // For each block instance that has the same definition, we need to keep track of the "maximum depth" at which is found.
+      // This will enable on receive to create them in the correct order (descending by max depth, interleaved definitions and instances).
+      // We need to interleave the creation of definitions and instances, as some definitions may depend on instances.
+      if (
+        !_instanceObjectsManager.TryGetInstanceProxiesFromDefinitionId(
+          definitionId.ToString(),
+          out List<InstanceProxy> instanceProxiesWithSameDefinition
+        )
+      )
+      {
+        instanceProxiesWithSameDefinition = new List<InstanceProxy>();
+        _instanceObjectsManager.AddInstanceProxiesByDefinitionId(
+          definitionId.ToString(),
+          instanceProxiesWithSameDefinition
+        );
+      }
+
+      // We ensure that all previous instance proxies that have the same definition are at this max depth. I kind of have a feeling this can be done more elegantly, but YOLO
+      foreach (var instanceProxyWithSameDefinition in instanceProxiesWithSameDefinition)
+      {
+        if (instanceProxyWithSameDefinition.maxDepth < depth)
+        {
+          instanceProxyWithSameDefinition.maxDepth = depth;
+        }
+      }
+
+      instanceProxiesWithSameDefinition.Add(_instanceObjectsManager.GetInstanceProxy(instanceId));
+
+      if (
+        _instanceObjectsManager.TryGetInstanceDefinitionProxy(
+          definitionId.ToString(),
+          out InstanceDefinitionProxy value
+        )
+      )
+      {
+        int depthDifference = depth - value.maxDepth;
+        if (depthDifference > 0)
+        {
+          // all MaxDepth of children definitions and its instances should be increased with difference of depth
+          _instanceObjectsManager.UpdateChildrenMaxDepth(value, depthDifference);
+        }
+        return;
+      }
+
+      var definition = (BlockTableRecord)transaction.GetObject(definitionId, OpenMode.ForRead);
+      var definitionProxy = new InstanceDefinitionProxy()
+      {
+        applicationId = definitionId.ToString(),
+        objects = new(),
+        maxDepth = depth,
+        name = !instance.AnonymousBlockTableRecord.IsNull ? "Dynamic instance " + definitionId : definition.Name,
+        ["comments"] = definition.Comments
```

Comment:
What are comments. Should this by typed?

Reply:
it is probably acad<>acad interop and we do not bother with typing it for `InstanceDefinitionProxy`, which is too general so far

Reply:
jsut checked and funny thing is that we don't consume this data on receive. need to check with @didimitrie